homeassistant/components/signal_messenger/notify.py

```diff
@@ -1,23 +1,30 @@
 """Signal Messenger for notify component."""
+from __future__ import annotations
+
 import logging
+from typing import Any
 
 from pysignalclirestapi import SignalCliRestApi, SignalCliRestApiError
+import requests
 import voluptuous as vol
 
 from homeassistant.components.notify import (
     ATTR_DATA,
     PLATFORM_SCHEMA,
     BaseNotificationService,
 )
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
 _LOGGER = logging.getLogger(__name__)
 
 CONF_SENDER_NR = "number"
 CONF_RECP_NR = "recipients"
 CONF_SIGNAL_CLI_REST_API = "url"
-ATTR_FILENAME = "attachment"
+CONF_MAX_ALLOWED_DOWNLOAD_SIZE_BYTES = 52428800
```

Comment:
Add this config to the `PLATFORM_SCHEMA` with this as the default? Also would need to document it.

Reply:
I think the name of this const is a bit misleading - it's not really a config/PLATFORM_SCHEMA property. I'll update the name to `ATTACHMENT_SIZE_LIMIT` in the next commit.

@burnsie-la can you please update your [docs PR](https://github.com/home-assistant/home-assistant.io/pull/20762) to mention the limit

Reply:
[Seems that the limit is 100mb](https://www.reddit.com/r/signal/comments/astgm2/comment/egxaq6i/). From my testing it seems that it is 100mb per file, not per message. I was able to upload files totalling over 100mb in a single message. Can this be enforced instead of the 50mb limit? Additionally can it be enforced on local files? Perhaps also just pass the local files as bytes to the lib along with url attachments.

Reply:
The 100MB limit that you linked to is the Signal service limit. We added this limitation to prevent causing issues with the HA server and/or Signal Docker container in the event that you inadvertently attempt to attach a massive file.

50MB was chosen to keep the attachment limits consistent with the [Home Assistant companion app](https://companion.home-assistant.io/docs/notifications/notification-attachments#supported-media-types).

Reply:
This PR is to add support for Attachments as Bytes from URLs.  We can revisit the local file handling in a future PR.

Reply:
I see but this isn't the companion app. It's a separate service with its own limits.